algorithms/Steamroller.js

```diff
@@ -0,0 +1,20 @@
+// Flatten a nested array. You must account for varying levels of nesting.
+
+
+function steamrollArray(arr) {
+  // I'm a steamroller, baby
+  var flattenArray = [];
+  while(arr.length > 0){
+    if(Array.isArray(arr[0]) === false){
+     flattenArray.push(arr[0]);
+     arr.shift(0);
```

Comment:
Hey @jrob8577 I'd love to go further into this in person -- when you have a moment, may you walk through the process with me really quickly?